lambda-events/src/main/scala/feral/lambda/events/SQSEvent.scala

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feral.lambda.events
+
+import io.circe.Decoder
+
+import scala.util.Try
+
+// https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/aws-lambda/trigger/sqs.d.ts
+// https://docs.aws.amazon.com/lambda/latest/dg/invoking-lambda-function.html#supported-event-source-sqs
+
+final case class SQSEvent(
+    records: List[SQSRecord]
+)
+
+object SQSEvent {
+  implicit val decoder: Decoder[SQSEvent] =
+    Decoder.instance(_.get[List[SQSRecord]]("Records")).map(SQSEvent(_))
+}
+
+final case class SQSRecord(
+    messageId: String,
+    receiptHandle: String,
+    body: String,
+    attributes: SQSRecordAttributes,
+    messageAttributes: Map[String, SQSMessageAttribute],
+    md5OfBody: String,
+    eventSource: String,
+    eventSourceARN: String,
+    awsRegion: String
+)
+
+object SQSRecord {
+  implicit val decoder: Decoder[SQSRecord] = Decoder.instance(i =>
+    for {
+      messageId <- i.get[String]("messageId")
+      receiptHandle <- i.get[String]("receiptHandle")
+      body <- i.get[String]("body")
+      attributes <- i.get[SQSRecordAttributes]("attributes")
+      messageAttributes <- i.get[Map[String, SQSMessageAttribute]]("messageAttributes")
+      md5OfBody <- i.get[String]("md5OfBody")
+      eventSource <- i.get[String]("eventSource")
+      eventSourceARN <- i.get[String]("eventSourceARN")
+      awsRegion <- i.get[String]("awsRegion")
+    } yield SQSRecord(
+      messageId,
+      receiptHandle,
+      body,
+      attributes,
+      messageAttributes,
+      md5OfBody,
+      eventSource,
+      eventSourceARN,
+      awsRegion
+    ))
+}
+
+final case class SQSRecordAttributes(
+    awsTraceHeader: Option[String],
```

Comment:
@bpholt Now this is interesting!

Reply:
Hmm, a subtlety I missed is that this trace header is not on the event itself, but rather on each individual record in the event.